discovery/discovery_test.go

```diff
@@ -756,3 +744,110 @@ func TestCachedPool_GetOrchestrators_TicketParamsValidation(t *testing.T) {
 	assert.Len(infos, 0)
 	sender.AssertNumberOfCalls(t, "ValidateTicketParams", 50)
 }
+
+func TestNewWHOrchestratorPoolCache(t *testing.T) {
+	require := require.New(t)
+	assert := assert.New(t)
+
+	// mock webhook and orchestrator info request
+	addresses := []string{"https://127.0.0.1:8936", "https://127.0.0.1:8937", "https://127.0.0.1:8938"}
+
+	getURLsfromWebhook = func(cbUrl *url.URL) ([]byte, error) { return []byte{}, nil }
```

Comment:
If we returned actual json here, then wouldn't need to stub out the `deserialzeWebhookJSON` part. That's somewhat preferable since it'll exercise more of the actual code paths. Something like:

```golang
getURLSfromWebhook = func ... {
  a := []
  for _, v := range addresses {
    append(a, webhookResponse{Address: a})
  }
  return json.Marshal(&a)
}
```

Reply:
Goooood point!! Thanks for pointing out this optimization